refactor(properties): make management common for substations and loads

TheMaskedTurtle opened the pull request.

dbraquart reviewed (changes requested):
code: ok, but headers missing.
tests: ko

Issues:
1) create a substation by copy:
ex: try to clone TRAN5 from a THTHT.It comes with 2 props paysCvg/regionCvg. When we realize, these 2 props are not created, and not retrieved when we edit the modification with the form.
In fact properties are not sent in the POST request.
It looks like a regression cause it's currently ok on dev-opf.
2) same issue with load by-copy creation
3) edit a substation (or load) having properties. Mark one property as deleted. save and re-open: we cannot validate a modification without having an error on marked property value (regression from dev-opf)
4) predefined codeOI property is not added seemingly (PR/MR in deployment repositories). See:
- Prod(SNP): <devin>gridsuite-deployment/-/blob/main/components/env-production/apps-metadata.json
- OPF: maybe same ?
- docker/dev: docker-compose/apps-metadata.json

dbraquart commented:
I tested codeOI configuration in my docker-compose envt. see [apps-metadata-codeoi-diff.txt](https://github.com/gridsuite/gridstudy-app/files/13965361/apps-metadata-codeoi-diff.txt)

TheMaskedTurtle commented on src/components/dialogs/network-modifications/common/properties/properties-form.tsx:
```diff
@@ -0,0 +1,123 @@
+import { GridSection } from '../../../dialogUtils';
```
Done

TheMaskedTurtle commented on src/components/dialogs/network-modifications/common/properties/property-utils.ts:
```diff
@@ -0,0 +1,181 @@
+/**
```
Done

TheMaskedTurtle commented:
> code: ok, but headers missing. tests: ko
> 
> Issues:
> 
>     1. create a substation by copy:
>        ex: try to clone TRAN5 from a THTHT.It comes with 2 props paysCvg/regionCvg. When we realize, these 2 props are not created, and not retrieved when we edit the modification with the form.
>        In fact properties are not sent in the POST request.
>        It looks like a regression cause it's currently ok on dev-opf.
> 
>     2. same issue with load by-copy creation
> 
>     3. edit a substation (or load) having properties. Mark one property as deleted. save and re-open: we cannot validate a modification without having an error on marked property value (regression from dev-opf)
> 
>     4. predefined codeOI property is not added seemingly (PR/MR in deployment repositories). See:
> 
> 
>     * Prod(SNP): gridsuite-deployment/-/blob/main/components/env-production/apps-metadata.json
> 
>     * OPF: maybe same ?
> 
>     * docker/dev: docker-compose/apps-metadata.json

@dbraquart thanks for your careful review.

1. Fixed
2. Fixed
3. I think the new behavior is more consistent with what is allowed in creation for example, to not allow to validate empty values. The problem here is more that we don't load the equipment value as value but only as previousValue, what do you think ? 
4. Yes I will add PRs for that

Mathieu-Deharbe commented on src/components/dialogs/network-modifications/common/properties/property-utils.ts:
```diff
@@ -0,0 +1,190 @@
+                        [VALUE]: null,
+                        [PREVIOUS_VALUE]: value,
+                        [DELETION_MARK]: false,
+                        [ADDED]: false,
```
Here we create a newModificationProperty from an equipment property and [ADDED] is set to false. While in getPropertiesFromEquipment [ADDED] is set to true. I don't understand this difference. Does ADDED mean added to the modification hypothesis ?

TheMaskedTurtle replied:
But I did the mistake previously, I thought we could always read "equipment properties" and return the same "modification properties" for both modification and creation dialogs but actually we couldn't.

Mathieu-Deharbe commented on src/components/dialogs/network-modifications/load/modification/load-modification-dialog.js:
```diff
@@ -96,6 +112,61 @@ const LoadModificationDialog = ({
+                    true
+                )
+                    .then((load) => {
+                        if (load) {
+                            setLoadToModify(load);
+                            reset((formValues) => ({
+                                ...formValues,
+                                [ADDITIONAL_PROPERTIES]:
+                                    getConcatenatedProperties(load),
+                            }));
+                        }
+                        setDataFetchStatus(FetchStatus.SUCCEED);
```
So if **!load,** FetchStatus is set to SUCCEED and setLoadToModify is not called ? So the displayed data remains unchanged ?

TheMaskedTurtle replied:
Actually, we do a backendFetch, that does a safeFetch that will throw if the HTTP response is not in 20X, and on this endpoint of study-server it also throws if the load is not found so HTTP response 400-599, sooo if (load) should actually always be true but I couldn't agree more that this conclusion is not straightforward. If we turn stuff to typescript at least we will have a better understanding of stuff like that.

Mathieu-Deharbe reviewed (approved):
Nice refacto !

Sorry to take so long, this javascript is a bit complicated for me, and I wanted to understand event the displaced code.

TheMaskedTurtle commented on src/components/dialogs/network-modifications/common/properties/property-utils.ts:
```diff
@@ -0,0 +1,181 @@
+    return metadata.name === 'Study';
+};
+
+export const fetchPredefinedProperties = (
+    networkElementType: string
+): Promise<PredefinedProperties | undefined> => {
+    return fetchAppsAndUrls().then((res: [Metadata]) => {
```
Discussing with Jon I understood that on the project we don't want to use much of async/await syntax

TheMaskedTurtle commented on src/components/dialogs/network-modifications/common/properties/property-utils.ts:
```diff
@@ -0,0 +1,181 @@
+    }
+    const validValues = properties.filter((v) => v?.name && v?.value);
+    return validValues.length === new Set(validValues.map((v) => v.name)).size;
+};
```
I did a bit of changes here, I renamed checkUniquePropertyNames

dbraquart reviewed (changes requested):
code: a change request for a missing space !
tests: all good to me
+ I have a suggestion : could we have a more accurate error message about property uniqueness ? adding than only the names are unique:
"DuplicatedProps": "Duplicated properties: each property name must be unique",
"DuplicatedProps": "Propriétés dupliquées : chaque nom de propriété doit être unique",
